server/src/main/java/org/opensearch/search/querygroup/QueryGroupLevelResourceUsageView.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.querygroup;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.search.resourcetypes.ResourceType;
+import org.opensearch.tasks.Task;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the point in time view of resource usage of a QueryGroup and
+ * has a 1:1 relation with a QueryGroup.
+ * This class holds the QueryGroup ID, the resource usage data, and the list of active tasks.
+ */
+@ExperimentalApi
+public class QueryGroupLevelResourceUsageView {
+
+    private final String queryGroupId;
+    // resourceUsage holds the resource usage data for a QueryGroup at a point in time
+    private final Map<ResourceType, Long> resourceUsage;
+    // activeTasks holds the list of active tasks for a QueryGroup at a point in time
+    private final List<Task> activeTasks;
+
+    public QueryGroupLevelResourceUsageView(String queryGroupId) {
+        this.queryGroupId = queryGroupId;
+        this.resourceUsage = new HashMap<>();
+        this.activeTasks = new ArrayList<>();
+    }
+
+    public QueryGroupLevelResourceUsageView(String queryGroupId, Map<ResourceType, Long> resourceUsage, List<Task> activeTasks) {
+        this.queryGroupId = queryGroupId;
+        this.resourceUsage = resourceUsage;
+        this.activeTasks = activeTasks;
+    }
+
+    /**
+     * Returns the resource usage data.
+     *
+     * @return The map of resource usage data
+     */
+    public Map<ResourceType, Long> getResourceUsageData() {
+        return resourceUsage;
+    }
+
+    /**
+     * Returns the list of active tasks.
+     *
+     * @return The list of active tasks
+     */
+    public List<Task> getActiveTasks() {
+        return activeTasks;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        QueryGroupLevelResourceUsageView that = (QueryGroupLevelResourceUsageView) o;
+        return Objects.equals(queryGroupId, that.queryGroupId);
```

Comment:
Why are we not comparing other fields of this object here ?

Reply:
Since a QueryGroupLevelResourceUsageView is unique to a queryGroup and queryGroupId is unique

Reply:
since queryGroupId would uniquely identify a QueryGroupLevelResourceUsageView as they have a 1:1 mapping

Reply:
Discussed offline with Kaushal 

ResourceType is a enum so it does not need hashcode and equals implemented.
But Task does not have an equals or hashcode implemented and i dont wanna touch those class
Hence sticking with this only for now.